### PR TITLE
Closes #352

### LINKED ISSUE #352: Update opacity management in actor.sphere
After looking more in detail into your fix, I think we should avoid using `actor.GetProperty().SetOpacity(opacity)`. It seems not compatible with our architecture. 
Ideally, in the function `actor.sphere`, we would check if `opacity != 1`.  if yes, we should update the color array by adding a dimension if it is an Nx3, or by updating the last dimension if it is an Nx4.

I will merge your PR because it fixes the `actor.point` but we need a new PR to improve this version.

_Originally posted by @skoudoro in https://github.com/fury-gl/fury/issues/351#issuecomment-755368331_

### PR DESCRIPTION
**Pull request by tantheta01**
Updated Opacity Management in Actor Sphere. The function actor.sphere now accepts a numpy array or a float as the opacity so different spheres of an actor can have different opacities.

### CODE REVIEW

**Comment by Nibba2018** [comment]
Hello @tantheta01 , Thank you for creating this PR. Looks like your PR contains **irrelevant** changes that you might have done to do some testing of your own. To fix that I would recommend you to create a new branch from fury-gl/master and then use that to make a pull request instead of using the master branch of your fork.

Finally, you need to write some tests in `fury/tests/test_actor.py` to check if your changes are working correctly and then I will review your PR.

Thanks.

File: fury/actor.py [hunk 1]
```diff
@@ -1462,17 +1462,37 @@ def sphere(centers, colors, radii=1., phi=16, theta=16,
...
+            colors = np.concatenate((colors, opacity), axis=1)
+        elif colors.shape[1]==4:
+            colors[:, 3] = opacity
+    else:
+        if colors.shape[1]==3:
+            colors = np.concatenate((colors, opacity.reshape(-1,1)), axis=1)
+        if colors.shape[1]==4:
+            colors[:, 3] = opacity.reshape(-1,)
+
+
```
Comment (skoudoro) [inline fury/actor.py:1484]: This code seems relevant and good. It will be good to add a test to show it!

File: fury/actor.py [hunk 2]
```diff
@@ -1437,8 +1437,8 @@ def sphere(centers, colors, radii=1., phi=16, theta=16,
     faces : ndarray, shape (M, 3)
         If faces is None then a sphere is created based on theta and phi angles
         If not then a sphere is created with the provided vertices and faces.
-    opacity : float, optional
-        Takes values from 0 (fully transparent) to 1 (opaque). Default is 1.
+    opacity : float or ndarray shape (N,)
```
Comment (skoudoro) [inline fury/actor.py:1440]: `float or ndarray shape (N,), optional`

File: fury/actor.py [hunk 3]
```diff
@@ -2553,23 +2573,26 @@ def sdf(centers, directions=(1, 0, 0), colors=(1, 0, 0), primitives='torus',
...
 
     verts, faces = fp.prim_box()
+    print("The size of verts and faces is ---", verts.shape, faces.shape)
     repeated = fp.repeat_primitive(verts, faces, centers=centers,
                                    colors=colors, directions=directions,
                                    scales=scales)
+   # print("Repeated is------", repeated)
 
     rep_verts, rep_faces, rep_colors, rep_centers = repeated
+    print("The size of following is : rep_verts", rep_verts.shape, "rep_faces", rep_faces.shape, "rep_colors", rep_colors.shape, "rep_centers", rep_centers.shape)
```
Comment (skoudoro) [inline fury/actor.py:2595]: many `print` to remove

**Review by skoudoro — COMMENTED** [review]
Hi @tantheta01, 

Thank you for this. Can you clean a bit your code and add a test? it looks like a part of your code should go to a future PR.

Thank you

Thank you

File: fury/actor.py (hunk 3 above)
Comment (tantheta01) [reply fury/actor.py:2595]: Hello @skoudoro 
I apologize for the mistakes and the inconvenience caused. I have created a fresh pull request with cleaned code and included unit tests. 
Thank You

**Comment by skoudoro** [comment]
closing in favor of #365

**Pull request closed without merging**